web/pdf_viewer.js

```diff
@@ -1868,14 +1868,18 @@ class PDFViewer {
       this._spreadMode !== SpreadMode.NONE &&
       this._scrollMode !== ScrollMode.HORIZONTAL;
 
-    // If we are scrolling and the rendering of a detail view was just
-    // cancelled, it's because the user is scrolling too quickly and so
-    // we constantly need to re-render a different area.
-    // Don't attempt to re-render it: this will be done once the user
-    // stops scrolling.
     const ignoreDetailViews =
-      this.#scrollTimeoutId !== null &&
-      visiblePages.views.some(page => page.detailView?.renderingCancelled);
+      // If we are scrolling and the rendering of a detail view was just
+      // cancelled, it's because the user is scrolling too quickly and so
+      // we constantly need to re-render a different area.
+      // Don't attempt to re-render it: this will be done once the user
+      // stops scrolling.
+      (this.#scrollTimeoutId !== null &&
+        visiblePages.views.some(page => page.detailView?.renderingCancelled)) ||
+      // If we are zooming, do not re-render the detail views. Re-renders on
+      // zoom happen with a delay, and once the rendering happens it will also
+      // trigger rendering of the detail views.
+      this.#scaleTimeoutId !== null;
```

Comment:
*Most likely a pointless micro-optimization, so please feel free to ignore this, but I wanted to ask:*
Would it make any sense to flip these conditions, and check `#scaleTimeoutId` first, since that may avoid the iteration above in some cases?